src/pineko/check.py

```diff
@@ -41,3 +41,29 @@ def check_grid_and_eko_compatible(pineappl_grid, operators):
     # x-grid
     if not np.all(in1d(np.unique(operators["targetgrid"]), np.array(x_grid))):
         raise ValueError("x grid in pineappl grid and eko operator are NOT compatible!")
+
+
+def check_grid_contains_sv(pineappl_grid, theory_card):
+    """
+    Raises a `ValueError if the theory_card asks for scale-variations but they are not
+    available in the pineappl grid.
+
+    Parameters
+    ----------
+        pineappl_grid : pineappl.grid.Grid
+            grid
+        theory_card : dict
+            theory card
+    """
+    xir = theory_card["XIR"]
+    xif = theory_card["XIF"]
+    ftr = theory_card["fact_to_ren_scale_ratio"]
```

Comment:
Don't propagate the `theory_card` too much, take these three elements as input of the function itself (it is more transparent, you can document it better, and use without the need of an actual `theory_card`).